test/source/tests/settings.ts

```diff
@@ -425,6 +460,29 @@ export let defineSettingsTests = (testVariant: TestVariant, testWithBrowser: Tes
       await settingsPage1.close();
     }));
 
+    ava.default('settings - email change', testWithBrowser('ci.tests.gmail', async (t, browser) => {
+      const acct1 = 'ci.tests.gmail@flowcrypt.test';
+      const acct2 = 'user@forbid-storing-passphrase-org-rule.flowcrypt.test';
+      const settingsPage = await browser.newPage(t, TestUrls.extensionSettings(acct1));
+      await SettingsPageRecipe.toggleScreen(settingsPage, 'additional');
+      const experimentalFrame = await SettingsPageRecipe.awaitNewPageFrame(settingsPage, '@action-open-module-experimental', ['experimental.htm']);
+      const { cryptup_citestsgmailflowcrypttest_passphrase_07481C8ACF9D49FE: savedPassphrase1 } =
+        await settingsPage.getFromLocalStorage(['cryptup_citestsgmailflowcrypttest_passphrase_07481C8ACF9D49FE']);
+      expect(savedPassphrase1).not.to.be.an('undefined');
+      await experimentalFrame.waitAndClick('@action-change-email');
+      const oauthPopup = await browser.newPageTriggeredBy(t, () => PageRecipe.waitForModalAndRespond(experimentalFrame, 'confirm',
+        { contentToCheck: 'email address has changed', clickOn: 'confirm' }));
+      await OauthPageRecipe.mock(t, oauthPopup, acct2, 'override_acct');
+      await PageRecipe.waitForModalAndRespond(experimentalFrame, 'confirm',
+        { contentToCheck: 'email from ci.tests.gmail@flowcrypt.test to user@forbid-storing-passphrase-org-rule.flowcrypt.test', clickOn: 'confirm' });
+      await PageRecipe.waitForModalAndRespond(experimentalFrame, 'info',
+        { contentToCheck: 'Email address changed to user@forbid-storing-passphrase-org-rule.flowcrypt.test', clickOn: 'confirm' });
```

Comment:
Actually, clicking 'confirm' here opens a new tab with settings. Did you mean we should apply new OrgRules there, @tomholub ?

Reply:
See my comment below. I'm personally not too fussed about how OrgRules are applied when domain of the user changes, however if you want to be the most correct, you can follow https://github.com/FlowCrypt/flowcrypt-browser/pull/3707#issuecomment-856996298